### DISCUSSION TITLE
Archiving SharePoint Online Site

### DISCUSSION THREAD

**Discussion by EdAlexander** [pnp/pnpcore#1507] [Q&A] [answered]
I am trying to find a pnp.core code equivelant to the PowerShell commandlet "Set-PNPSiteArchiveState".  Is there a supported way to do this in c#?  

I have tried simple restful call with bearer token but always get "unauthorized" as response.
`            using (var sourceContext = context.Clone(new System.Uri(SelectedCase.caseUrl)))
            {
                
                using (var httpClient = new HttpClient())
                {
                   
                    var accessToken = await context.AuthenticationProvider
                        .GetAccessTokenAsync(new Uri("https://{mytennant}.sharepoint.com/.default"));
                    
                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);

                    var requestUri = new Uri("https://{mytennant}-admin.sharepoint.com/_api/SiteArchive/Archive");
                    var content = new StringContent("{\"siteUrl\":\"https://{mytennant}.sharepoint.com/sites/" + sourceSite + "\"}", System.Text.Encoding.UTF8, "application/json");

                    var response = await httpClient.PostAsync(requestUri, content);

                    if (response.IsSuccessStatusCode)
                    {
                        Console.WriteLine("Site archived successfully.");
                    }
                    else
                    {
                        Console.WriteLine($"Failed to archive site. Status code: {response.StatusCode}");
                    }
                }

**Comment by jansenbe** [comment, marked as answer]
@EdAlexander : we would need to either implement the needed REST calls or "wrap" the relevant CSOM commands Tenant.ArchiveSite and Tenant.UnArchiveSite (https://github.com/pnp/powershell/blob/43dc32f39938814c7581706fe8de2b7899a2e61c/src/Commands/Admin/SetSiteArchiveState.cs#L12) into the Admin library to realize this feature. I'll transfer this to the issue list to track that ask.

**Reply by EdAlexander** [reply]
Thanks for the update!  It would be fantastic to get the function available.
